### ISSUE TITLE
training on multi-gpu

### ISSUE THREAD

**Issue by harshit4084** [zhangjunh/DR-GAN-by-pytorch#11]
hello,
I want to run this code on multi-gpu, 
can you tell what minor or major changes I need to do if you have done so earlier?

**Comment by zhangjunh** [comment]
Hi @harshit4084 , this project was finished last year and I didn't use multi-gpu when I code it. I'm not sure how to run this code on multi-gpu, so I couldn't help you. If I have the time in the future I will add it to the code, and now you might make it on your own.

**Comment by harshit4084** [comment]
Hi @zhangjunh Yes. I converted the code for multi-gpu. And it is working fine for single DR GAN and giving some dim error when splitting on multiple gpus in Multi DRGAN.
I'll put it here, you'll be able to interpret it better. 
Also, Approximately how much time did it take to train the model to give a decent output?

**Comment by zhangjunh** [comment]
Hi @harshit4084 , I couldn't remember the exact time for training munti-DRGAN. It's might be more than a week using single 1080TI.

**Comment by MmDawN** [comment]
> Hi @zhangjunh Yes. I converted the code for multi-gpu. And it is working fine for single DR GAN and giving some dim error when splitting on multiple gpus in Multi DRGAN.
> I'll put it here, you'll be able to interpret it better.
> Also, Approximately how much time did it take to train the model to give a decent output?

Hello, I am also trying to use multiple GPUs to run this code, but after `torch.nn.DataParallel processing` , the batchsize of the three variables that input to the Encoder is inconsistent. Have you ever encountered this problem? If possible, can you provide us with the multi-GPU running solution you have implemented, thank you!
